src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,311 @@
+import duke.command.DeleteCommand;
+import duke.command.MarkCommand;
+import duke.command.UnMarkCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+
+import java.time.format.DateTimeParseException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the part of the chatbot that deals with making sense of the command.
+ */
+public class Parser {
+    /**
+     * Returns a command that is specified in the user input.
+     * 
+     * @param command Command that the user keys in.
+     * @return the specified command in the user's input.
+     * @throws DukeException  If input is invalid.
+     */
+    public static Command parse(String command) throws DukeException {
+        if (isExitCommand(command)) {
+            return new ExitCommand();
+        } else if (isListCommand(command)) {
+            return new ListCommand();
+        } else if (isMarkCommand(command)) {
+            String input = command.replaceFirst("mark", "").trim();
+            int index = validateMarkIndex(input);
+            return new MarkCommand(index);
+        } else if (isUnMarkCommand(command)) {
+            String input = command.replaceFirst("unmark", "").trim();
+            int index = validateMarkIndex(input);
+            return new UnMarkCommand(index);
+        } else if (isAddToDoCommand(command)) {
+            String todo = command.replaceFirst("todo", "").trim();
+            validateToDo(todo);
+            return new AddToDoCommand(todo);
+        } else if (isAddDeadlineCommand(command)) {
+            String[] deadline = validateDeadline(command);
+            if (isDate(deadline[1])) {
+                LocalDateTime deadlineDate = parseDate(deadline[1]);
+                return new AddDeadlineCommand(deadline[0], deadlineDate);
+            } else {
+                return new AddDeadlineCommand(deadline[0], deadline[1]);
+            }
+        } else if (isAddEventCommand(command)) {
+            String[] event = validateEvent(command);
+            if (isDate(event[1])) {
+                LocalDateTime eventDate = parseDate(event[1]);
+                LocalDateTime endTime = parseDate(event[2]);
+                return new AddEventCommand(event[0], eventDate, endTime);
+            } else {
+                return new AddEventCommand(event[0], event[1], event[2]);
+            }
+        } else if (isDeleteCommand(command)) {
+            String input = command.replaceFirst("delete", "").trim();
+            int index = validateDeleteIndex(input);
+            return new DeleteCommand(index);
+        } else if (isFindCommand(command)) {
+            String keyword = command.replace("find", "").trim();
+            validateFindInput(keyword);
+            return new FindCommand(keyword);
+        } else {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
```

Comment:
This method is really long! You could try to refactor your code by breaking it down into multiple sub-methods for better readability.